src/Hipace.H

```diff
@@ -290,6 +290,7 @@ public:
      */
     void CheckGhostSlice (int it);
 
+    static int m_nlev; /**< number of MR levels */
```

Comment:
Class Hipace should inherit this type of data, you can use function `finestLevel()` too or, as you do, `maxLevel()`. For loops can remain as compact with
```c++
for (int lev = 0; lev <= finestLevel(); ++lev)
```
and of course `nLevel = finestLevel()+1`, so it's probably unnecessary to define this new member variable.

Reply:
Ok thanks, I will adapt this